tools/crm_resource_print.c

```diff
@@ -152,6 +152,57 @@ attribute_list_default(pcmk__output_t *out, va_list args) {
     return pcmk_rc_ok;
 }
 
+PCMK__OUTPUT_ARGS("agent-status", "int", "const char *", "const char *", "const char *",
+                  "const char *", "const char *", "int")
+static int
+agent_status_default(pcmk__output_t *out, va_list args) {
+    int status = va_arg(args, int);
+    const char *action = va_arg(args, const char *);
+    const char *name = va_arg(args, const char *);
+    const char *class = va_arg(args, const char *);
+    const char *provider = va_arg(args, const char *);
+    const char *type = va_arg(args, const char *);
+    int rc = va_arg(args, int);
+
+    if (status == PCMK_LRM_OP_DONE) {
+        out->info(out, "Operation %s%s%s (%s:%s:%s) returned: '%s' (%d)",
+                  action, name ? " for " : "", name ? name : "", class,
+                  provider ? provider : "", type,
+                  services_ocf_exitcode_str(rc), rc);
+    } else {
+        out->err(out, "Operation %s%s%s (%s:%s:%s) failed: '%s' (%d)",
+                 action, name ? " for " : "", name ? name : "", class,
+                 provider ? provider : "", type,
+                 services_lrm_status_str(status), status);
+    }
+
+    return pcmk_rc_ok;
+}
+
+PCMK__OUTPUT_ARGS("agent-status", "int", "const char *", "const char *", "const char *",
+                  "const char *", "const char *", "int")
+static int
+agent_status_xml(pcmk__output_t *out, va_list args) {
+    int status G_GNUC_UNUSED = va_arg(args, int);
+    const char *action G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *name G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *class G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *provider G_GNUC_UNUSED = va_arg(args, const char *);
+    const char *type G_GNUC_UNUSED = va_arg(args, const char *);
+    int rc = va_arg(args, int);
+
+    char *status_str = pcmk__itoa(rc);
+
+    pcmk__output_create_xml_node(out, "agent-status",
+                                 "code", status_str,
+                                 "message", services_ocf_exitcode_str(rc),
+                                 NULL);
```

Comment:
I think the other arguments would be useful to know in XML too

Reply:
The others are included by the surrounding resource-agent-action element, so I didn't feel like it needed to be doubled up here.  Of course if I rearrange things for a resource-agent-action output message, I might have to change this.